sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuColumnVectorUtils.scala

```diff
@@ -38,4 +40,99 @@ object GpuColumnVectorUtils {
     columnVectors.asInstanceOf[Array[ColumnVector]]
   }
 
+  def isCaseWhenFusionSupportedType(dataType: DataType): Boolean = {
+    dataType match {
+      case BooleanType => true
+      case ByteType => true
+      case ShortType => true
+      case IntegerType => true
+      case LongType => true
+      case FloatType => true
+      case DoubleType => true
+      case StringType => true
+      case DecimalType() => true
```

Comment:
Do we only support `Decimal(10,0)`? Why? Shouldn't this be `case _: DecimalType => true`? Why are the tests for DecimalType passing? Or are they not passing?

Reply:
Spark defines an `unapply` [method retuning only a boolean](https://github.com/apache/spark/blob/master/sql/api/src/main/scala/org/apache/spark/sql/types/DecimalType.scala#L248) as below for match test only,
```
def unapply(t: DataType): Boolean = t.isInstanceOf[DecimalType]
```
it is not an extractor but it makes `case DecimalType() => true` work. even it is not recommended.

Reply:
Both `DecimalType(15, 2)` and `DecimalType(10, 0)` match the branch: `case DecimalType() => true`

```scala
scala> import org.apache.spark.sql.types._
import org.apache.spark.sql.types._

scala>   def isCaseWhenFusionSupportedType(dataType: DataType): Boolean = {
     |     dataType match {
     |       case BooleanType => true
     |       case ByteType => true
     |       case ShortType => true
     |       case IntegerType => true
     |       case LongType => true
     |       case FloatType => true
     |       case DoubleType => true
     |       case StringType => true
     |       case DecimalType() => true
     |       case _ => false
     |     }
     |   }
isCaseWhenFusionSupportedType: (dataType: org.apache.spark.sql.types.DataType)Boolean

scala> isCaseWhenFusionSupportedType(DecimalType(15, 2))
res4: Boolean = true

scala> isCaseWhenFusionSupportedType(DecimalType(10, 0))
res5: Boolean = true

```

Reply:
Changed to : `case _: DecimalType => true`